src/graph/ScanExecutor.cpp

```diff
@@ -0,0 +1,349 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "graph/ScanExecutor.h"
+#include "base/Base.h"
+#include "time/WallClock.h"
+#include "utils/NebulaKeyUtils.h"
+
+namespace nebula {
+namespace graph {
+
+ScanExecutor::ScanExecutor(Sentence *sentence, ExecutionContext *ectx)
+    : TraverseExecutor(ectx, "scan") {
+    sentence_ = static_cast<ScanSentence *>(sentence);
+}
+
+static Status _getInt(Expression* expr, int64_t &v) {
+    Getters getters;
+
+    auto status = expr->prepare();
+    if (!status.ok()) {
+        return status;
+    }
+    auto value = expr->eval(getters);
+    if (!value.ok()) {
+        return status;
+    }
+    auto r = value.value();
+    if (!Expression::isInt(r)) {
+        return Status::Error("should be of type integer");
+    }
+    v = Expression::asInt(r);
+
+    return Status::OK();
+}
+
+Status ScanExecutor::prepare() {
+    auto space = ectx()->rctx()->session()->space();
+
+    auto tag = sentence_->tag();
+
+    auto tagStatus = ectx()->schemaManager()->toTagID(space, tag);
+    if (!tagStatus.ok()) {
+        LOG(ERROR) << "No schema found for " << tag;
+        return Status::Error("No schema found for `%s'", tag.c_str());
+    }
+    auto tagId = tagStatus.value();
+    return_columns_[tagId] = {};
+
+    {
+        auto expr = sentence_->partition();
+        int64_t partitionId = 0;
+        auto status = _getInt(expr, partitionId);
+        if (!status.ok()) {
+            return status;
+        }
+        partition_ = partitionId;
+    }
+
+    {
+        auto expr = sentence_->from();
+        if (expr) {
+            VertexID vertexId = 0;
+            auto status = _getInt(expr, vertexId);
+            if (!status.ok()) {
+                return status;
+            }
+            cursor_ = NebulaKeyUtils::vertexPrefix(partition_, vertexId);
+        }
+    }
+
+    {
+        auto expr = sentence_->latestSeconds();
+        if (expr) {
+            int64_t latestSeconds = 0;
+            auto status = _getInt(expr, latestSeconds);
+            if (!status.ok()) {
+                return status;
+            }
+            auto now = time::WallClock::fastNowInMicroSec();
+            startTime_ = now - latestSeconds * 1000000L;
+            endTime_ = now;
+        } else {
+            startTime_ = 0;
+            endTime_ = std::numeric_limits<int64_t>::max();
+        }
+    }
+
+    {
+        auto expr = sentence_->limit();
+        if (expr) {
+            int64_t limit = 0;
+            auto status = _getInt(expr, limit);
+            if (!status.ok()) {
+                return status;
+            }
+            limit_ = limit;
+        } else {
+            limit_ = std::numeric_limits<int32_t>::max();
```

Comment:
The default limit need to be reconsidered. 
It is dangerous with no limits.  Because as your logic,  all data will been scanned out,  it will block the normal OLTP requests.  Meanwhile,  on graphd-side,  it usually has not enough memory for the response.

Reply:
make sense.